src/api/orgAuthPlugin.ts

```diff
@@ -0,0 +1,116 @@
+import { getModelForClass } from '@typegoose/typegoose';
+import envVar from 'env-var';
+import type { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import fastifyPlugin, { type PluginMetadata } from 'fastify-plugin';
+import type { Connection } from 'mongoose';
+
+import { MemberModelSchema, Role } from '../models/Member.model.js';
+import { HTTP_STATUS_CODES } from '../utilities/http.js';
+import type { Result } from '../utilities/result.js';
+import type { PluginDone } from '../utilities/fastify/PluginDone.js';
+
+interface OrgRequestParams {
+  readonly orgName: string;
+  readonly memberId: string;
+}
+
+interface AuthenticatedFastifyRequest extends FastifyRequest {
+  user: { sub: string };
+}
+
+interface AuthorisedFastifyRequest extends AuthenticatedFastifyRequest {
+  isUserAdmin: boolean;
+}
+
+interface AuthorisationGrant {
+  readonly isAdmin: boolean;
+  readonly reason: string;
+}
+
+async function decideAuthorisation(
+  userEmail: string,
+  request: FastifyRequest,
+  dbConnection: Connection,
+  superAdmin?: string,
+): Promise<Result<AuthorisationGrant, string>> {
+  if (superAdmin === userEmail) {
```

Comment:
Maybe we can set multiple super admin accounts? We could have an array?

Reply:
Also, because we're delegating authentication to a third party, operators could easily configure their auth server to issue access tokens from different sets of user credentials.

For example, they could have one set of credentials to access the API programmatically via the JS library and another set of credentials to access the API manually via a CLI. Both sets of credentials would map to the same email address.